### PR TITLE
add hashicorp vault integration to website

### PR DESCRIPTION
**Pull request by UniserveJake**
# Details
* **Does this resolve an issue?**
No

### New Features
* Adds a hashicorp vault integration to the website

### Breaking Changes
* None

## Additional
Basic integration with hashicorp vault

### CODE REVIEW

File: website/integrations/services/hashicorp-vault/index.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+**Protocol Settings**
+
+- Name: Vault
+- Signing Key: Select any available key
+
+- Redirect URIs/Origins:
+```
+https://vault.company/ui/vault/auth/oidc/oidc/callback
+https://vault.company/oidc/callback
```
Comment (BeryJu) [inline website/integrations/services/hashicorp-vault/index.md:40]: not sure the second one is correct, I assume that's one for local CLI-based login? In that case it should be `http://localhost:8250/oidc/callback` imo

File: website/integrations/services/hashicorp-vault/index.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+         oidc_client_secret="Client Secret" \
+         default_role="reader"
+```
+
+Create the reader role
+```
+vault write auth/oidc/role/reader \
+      bound_audiences="$AUTH0_CLIENT_ID" \
+      allowed_redirect_uris="http://vault.company/ui/vault/auth/oidc/oidc/callback" \
+      allowed_redirect_uris="http://vault.company/oidc/callback" \
```
Comment (BeryJu) [inline website/integrations/services/hashicorp-vault/index.md:77]: Same as above

File: website/integrations/services/hashicorp-vault/index.md (hunk 1 above)
Comment (UniserveJake) [reply website/integrations/services/hashicorp-vault/index.md:40]: So, for the second one, Using vault from my laptop(not on the vault server) I had to set it to vault.company, It almost might be worth setting all three. Running vault commands on the vault server will definitely be worthwhile. I'll add it.

**Comment by BeryJu** [comment]
I forgot to mention, please add it to https://github.com/goauthentik/authentik/blob/master/website/sidebarsIntegrations.js then I can merge it.

**Pull request merged**